packages/grid_client/src/modules/gateway.ts

```diff
@@ -23,15 +27,41 @@ class GWModule extends BaseModule {
   workloadTypes = [WorkloadTypes.gatewayfqdnproxy, WorkloadTypes.gatewaynameproxy];
   gateway: GatewayHL;
 
+  /**
+   * Represents a module for managing gateway deployments.
+   *
+   * This class extends the `BaseModule` class and provides methods for deploying, listing, getting, and deleting gateway deployments.
+   * It includes functionality to deploy gateways with FQDN and name, check balance before deployment, and handle deployment updates.
+   * The module also supports retrieving specific gateway instances, deleting instances, and getting detailed information about deployments.
+   *
+   * @class GWModule
+   * @param {GridClientConfig} config - The configuration object for initializing the client.
+   */
   constructor(public config: GridClientConfig) {
     super(config);
     this.gateway = new GatewayHL(config);
   }
 
+  /**
+   * Deploys a gateway with FQDN.
+   *
+   * This method deploys a gateway with the provided FQDN configuration.
```

Comment:
it deploys a workload not a gateway
```suggestion
   * Deploys a gateway workload with FQDN.
   *
   * This method deploys a gateway workload with the provided FQDN configuration.
```